app/view/LayerTree.js

```diff
@@ -54,228 +55,33 @@ Ext.define('CpsiMapview.view.LayerTree', {
-     * based on the given tree structure loaded in #loadTreeStructure.
-     * For each folder an OL layer group is created and gets aggregated in a
-     * root layer group.
-     *
-     * @param  {Object} treeJson LayerTree structure
-     * @return {ol.layer.Group}  Root layer group
-     */
-    getGroupedLayers: function (treeJson) {
-        var me = this;
-
-        // wrapping all under the 'root' node aggregating all together
-        var rootLayerGroup = new ol.layer.Group({
-            name: 'root',
-            layers: []
-        });
-        // recursively create the OL layer group by the given tree structure
-        me.createOlLayerGroups(treeJson.children, rootLayerGroup);
-
-        return rootLayerGroup;
-    },
-
-    /**
-     * Creates recursively the OL layer groups for the given tree structure and
-     * puts them all together in the given parent group so they get folders in the LayerTree.
-     * Layers are directly put to the given parent group so they appear as "leafs" in the LayerTree.
+     * Updates the layer node UI for the given layer.
      *
-     * @param  {Object} treeNodesJson Child section of the LayerTree structure
-     * @param  {ol.layer.Group} parentGroup The parent group to put children (another groups / layers) into
+     * @param  {ol.layer.Base} layer The layer to update in the tree
      */
-    createOlLayerGroups: function (treeNodeChilds, parentGroup) {
-        var me = this;
-        // go over all passed in tree childs nodes
-        Ext.each(treeNodeChilds, function (child) {
-            // layer groups --> folders in tree
-            if (child.isLeaf !== true) {
-                // create empty layer group for this level
-                var layerGroup = new ol.layer.Group({
-                    name: child.title,
-                    layers: [],
-                });
-
-                var parentLayers = parentGroup.getLayers();
-                parentLayers.insertAt(0, layerGroup);
-
-                // recursion
-                me.createOlLayerGroups(child.children, layerGroup);
-            } else {
-                // layers --> leafs in tree
-                var mapLyr = BasiGX.util.Layer.getLayerBy('layerKey', child.id);
-                if (mapLyr) {
-                    parentGroup.getLayers().insertAt(0, mapLyr);
-                } else {
-                    Ext.Logger.warn('Layer with layerKey ' + child.id + ' not found in map layers');
-                }
-            }
-        });
+    updateLayerNodeUi: function (layer) {
+        var node = this.getNodeForLayer(layer);
+        node.triggerUIUpdate();
     },
 
     /**
-     * Updates the layer node UI for the given layer.
+     * Gets the layer node UI for the given layer.
      *
      * @param  {ol.layer.Base} layer The layer to update in the tree
      */
-    updateLayerNodeUi: function (layer) {
+    getNodeForLayer: function (layer) {
```

Comment:
Is there any reason not to also move this (and updateLayerNodeUi) into the controller?

Reply:
I think the `getNodeForLayer` is a view specific thing because it gets the view element (the node) corresponding to the layer. It is also called from a different component and it this way I was able to find it with `Ext.ComponentQuery`.

Reply:
Thanks - makes sense.